client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java

```diff
@@ -609,4 +612,170 @@ public void testRollover() throws IOException {
+            assertThat(getAliasesResponse.getAliases().get("index1").size(), equalTo(1));
+            assertThat(getAliasesResponse.getAliases().get("index1").get(0), notNullValue());
+            assertThat(getAliasesResponse.getAliases().get("index1").get(0).alias(), equalTo("alias1"));
+            assertThat(getAliasesResponse.getAliases().get("index2").size(), equalTo(1));
+            assertThat(getAliasesResponse.getAliases().get("index2").get(0), notNullValue());
+            assertThat(getAliasesResponse.getAliases().get("index2").get(0).alias(), equalTo("alias2"));
+            assertThat(getAliasesResponse.getAliases().get("index3").size(), equalTo(0));
+        }
+        {
+            GetAliasesRequest getAliasesRequest = new GetAliasesRequest();
+            GetAliasesResponse getAliasesResponse = execute(getAliasesRequest, highLevelClient().indices()::getAlias,
+                    highLevelClient().indices()::getAliasAsync);
+
+            assertThat(getAliasesResponse.getAliases().size(), equalTo(3));
+            assertThat(getAliasesResponse.getAliases().get("index1").size(), equalTo(1));
+            assertThat(getAliasesResponse.getAliases().get("index1").get(0), notNullValue());
+            assertThat(getAliasesResponse.getAliases().get("index1").get(0).alias(), equalTo("alias1"));
+            assertThat(getAliasesResponse.getAliases().get("index2").size(), equalTo(1));
+            assertThat(getAliasesResponse.getAliases().get("index2").get(0), notNullValue());
+            assertThat(getAliasesResponse.getAliases().get("index2").get(0).alias(), equalTo("alias2"));
+            assertThat(getAliasesResponse.getAliases().get("index3").size(), equalTo(0));
+        }
+    }
+
+    public void testGetAliasesNonExistentIndexOrAlias() throws IOException {
+
+        String alias = "alias";
+        String index = "index";
+        {
+            GetAliasesRequest getAliasesRequest = new GetAliasesRequest().indices(index);
+            ElasticsearchException exception = expectThrows(ElasticsearchException.class,
+                    () -> execute(getAliasesRequest, highLevelClient().indices()::getAlias, highLevelClient().indices()::getAliasAsync));
+            assertThat(exception.status(), equalTo(RestStatus.NOT_FOUND));
+            assertThat(exception.getMessage(), equalTo("Elasticsearch exception [type=index_not_found_exception, reason=no such index]"));
+        }
+        {
+            GetAliasesRequest getAliasesRequest = new GetAliasesRequest(alias);
+            ElasticsearchException exception = expectThrows(ElasticsearchException.class,
+                    () -> execute(getAliasesRequest, highLevelClient().indices()::getAlias, highLevelClient().indices()::getAliasAsync));
+            assertThat(exception.status(), equalTo(RestStatus.NOT_FOUND));
+            assertThat(exception.getMessage(), equalTo("Elasticsearch exception [type=exception, reason=alias [" + alias + "] missing]"));
```

Comment:
I am somehow stuck with this response in case of an error. I do not see an option which "makes sense".

A request with non existent alias results in : 

```
{
  "error": "alias [<alias>] missing",
  "status": 404
}
```

For now I decided to map it to an exception. Basically if there are *no* aliases found then throw. If at least one alias is found, a response will be returned and its status must be checked ( to verify if all aliases have been found : OK , or if this is a "partial" response : NOT_FOUND)

Another option is to throw only if any of the indices do not exist. And for the aliases always to return a response ( even if none of the specified aliases exist ).

Yet another option is to never throw... ( which is not is sync with the rest of the apis ... )

Ideas how to proceed ?

Reply:
no let's not wait, I would get this one in, for now let's never throw exception, users can read errors if they wish.

Reply:
@javanna one more thing: if ES exception is added to the response, should it be serialized on the transport layer ? Maybe it would be better not to serialize it on the transport layer, but then the response tests could get tricky and maybe they will need to be split ( something similar to what has been done in #28892 )

Reply:
I am not sure. At the moment we only have a string and a status, are you going for recreating an exception based on that info?